src/mlpack/tests/main_tests/dbscan_test.cpp

```diff
@@ -429,4 +429,27 @@ BOOST_AUTO_TEST_CASE(DBSCANNaiveSearchTest)
   CheckMatrices(output, naiveOutput);
 }
 
+/**
+ * Check that the random selection is same 
```

Comment:
Actually I think we should make this test a little bit more challenging.  Here the goal should be to ensure that `random_selection` affects the results.  So we should find a dataset that, when we set `random_selection = false`, the results are different than when we set `random_selection = true` (for the vast majority of random seeds).  Maybe you could make this as a synthetic dataset, or see if different parameters of `epsilon` or `minPoints` for the iris dataset can work.

Reply:
In fact, I needed a lot of advice on this part. Okay. I'll try to figure it out.

Reply:
Give it a shot and let me know if I can help out.  This may be a little tricky to get right.

Reply:
Yes, I'm currently working on this part. Thanks for the consideration!! :+1:

Reply:
Hi, @rcurtin . I found the optimal parameter for maximizing difference between two using iris dataset. I used 'Grid Search algorithm' to find the optimal parameter, epsilon and min_size. 
The range of values I checked is as follows.
Epsilon: 0.001 ~ 2 (interval: 0.001)
Min_size: 1 ~ 10 (interval: 1)
So I examined a total of 20,000 parameters, the maximum difference is 0.98 and is when epsilon is 0.358 and min_size is 1. And I think the A shows the same result always in iris dataset that I used. Thus, I will try to write a test code comparing A's result to B's. I want to ask you if this is a good approach. I'll contact you again when I'm done. 
Thanks. :)

Reply:
Yes, definitely `OrderedPointSelection` will show the same result in most places.  If you are finding that even with different random seeds, the results are always different, then that will work for the test.  Note that for anything in `main_tests/` (or more specifically where `BINDING_TYPE` is defined as `BINDING_TYPE_TEST`), the `math::RandomSeed()` function does nothing.  (This is because many mlpack bindings have an option for the random seed, and we don't want to set the random seed over and over again in the test for reproducibility.)

You can set the random seed by doing this:

```
    const size_t seed = std::time(NULL); // or some specific seed if you want
    randGen.seed((uint32_t) seed);
    srand((unsigned int) seed);
    arma::arma_rng::set_seed(seed);
```

That's just an adaptation of `math::RandomSeed()`.

Reply:
Oh, Thank you. I will test it.